packages/salesforce-adapter/src/client/user_facing_errors.ts

```diff
@@ -18,27 +18,27 @@ import { DeployMessage } from 'jsforce'
 import { decorators, values } from '@salto-io/lowerdash'
 import _ from 'lodash'
 import { logger } from '@salto-io/logging'
-import { ENOTFOUND, ERROR_HTTP_502, INVALID_GRANT, SF_REQUEST_LIMIT_EXCEEDED } from '../constants'
+import {
+  ENOTFOUND,
+  ERROR_HTTP_502,
+  ERROR_PROPERTIES,
+  INVALID_GRANT,
+  isSalesforceError, SALESFORCE_ERRORS,
+} from '../constants'
 
 const log = logger(module)
 const { isDefined } = values
 
-
-/**
- * To allow more robust support and avoid boilerplate
- * for supporting different Error objects (e.g. Node DNSError, JSForce error)
- * this values can be any of the Error object properties.
- */
-export const MAPPABLE_ERROR_PROPERTIES = [
+export const MAPPABLE_ERROR_NAME = [
   ERROR_HTTP_502,
-  SF_REQUEST_LIMIT_EXCEEDED,
+  SALESFORCE_ERRORS.REQUEST_LIMIT_EXCEEDED,
   INVALID_GRANT,
   ENOTFOUND,
 ] as const
 
-export type MappableErrorProperty = typeof MAPPABLE_ERROR_PROPERTIES[number]
+export type MappableErrorName = typeof MAPPABLE_ERROR_NAME[number]
 
-type MapErrorFunc = (error: Error) => string
+type GetUserFriendlyErrorFunc = (error: Error) => string | undefined
```

Comment:
Also not sure why this function can return undefined, I'm not sure this is how I would have done this type safety. The way I would have modeled this is probably. Let's discuss a bit f2f and see if we can model this a bit differently
```suggestion
type UserFriendlyError = INVALID_GRANT_ERROR | ....
type GetUserFriendlyErrorFunc<T extends UserFriendlyError> = (error: T) => string
```

Reply:
Changed the design